### PR TITLE
Let code owners merge PRs

### PR DESCRIPTION
**Pull request by Eirenliel**
Adds CODEOWNERS file and assigns some codeownders. The codeowners can merge PRs by posting `LGTM` if all checks passed using https://github.com/marketplace/actions/code-owner-self-merge?version=1.6.5

### CODE REVIEW

**Comment by Eirenliel** [comment]
suggestions for code owners are welcomed

**Comment by Louka3000** [comment]
I think @loucass003 should be code owner for server config since he wrote all of it: `server/src/main/java/dev/slimevr/config`

and I'd like to be code owner on skeleton stuff: `server/src/main/java/dev/slimevr/vr/processor` (which is changing to `server/src/main/java/dev/slimevr/tracking/processor` with my refactor...) and tracker filtering: `server/src/main/java/dev/slimevr/filtering`

**Comment by Eirenliel** [comment]
👍

File: .github/CODEOWNERS
Comment (TheButlah) [inline .github/CODEOWNERS:23]:
```suggestion
server/src/main/java/dev/slimevr/config/ @loucass003

```

File: .github/workflows/codeowners-merge.yml
Comment (TheButlah) [inline .github/workflows/codeowners-merge.yml:18]:
```suggestion
          if_no_maintainers_assign: '@Eirenliel @ButterscotchV'
```

**Review by Louka3000 — DISMISSED** [review]

**Review by ButterscotchV — APPROVED** [review]

File: .github/CODEOWNERS
Comment (ImUrX) [inline .github/CODEOWNERS:7]:
```suggestion

# Rust part of the GUI
/gui/src-tauri/ @ImUrX @TheButlah 

```

**Review by ImUrX — CHANGES_REQUESTED** [review]

**Comment by Louka3000** [comment]
_Actually, why does Butter get full merge perms on all of /server, shouldn’t I also do… I even wrote more code :nya_a:_

**Comment by Louka3000** [comment]
Important stuff from Discord:

With the current way it is, I won’t be able to do anything except hotfixes to these specific packages, since implementing anything new would require config or protocol/rpc changes. 

So, for example, even if 99% of my code in a PR to add a new skeleton config toggle is in the skeleton package, if it has a line of code in RPCSettingsHandler to make it work with the protocol, it’d absolutely need Butter or Eiren to merge…

Something like https://github.com/marketplace/actions/auto-request-review would allow to automatically request reviews when a PR touches a certain part of the code.
I propose Butter and I have merge perms for server, and moving the specific perms (Autobone, BVH, Skeleton, OSC and Filtering) to auto-reviews.

Btw, having perms to merge `gui/src/components/settings` would be nice too, since I’ve been touching that part a lot

**Review by Louka3000 — COMMENTED** [review]
^

**Comment by Eirenliel** [comment]
> The problem thus is that my codeowner perms are useless except for small hotfixes inside given packages, as any change that brings new stuff to the server will need code change outside of those packages.

We were talking about letting you to merge language changes in the first place. This system is a workaround from giving permissions to merge on everything. I'm trying to make a "well you can sort at least some of it without me and i'll still sleep well" kind of system.

**Comment by Louka3000** [comment]
My original proposal was just to be able to merge on x approving reviews.
Here, you’re giving Butter ownership of all server code, which means we could already merge full-fledged features or substantial code changes, but why only Butter? Feels unfair to me considering most recent server code is mine…

**Review by Louka3000 — APPROVED** [review]

**Comment by Eirenliel** [comment]
> With the current way it is, I won’t be able to do anything except hotfixes to these specific packages, since implementing anything new would require config or protocol/rpc changes.

You shouldn't review and merge your own PRs either way. This is for when someone else changes your code, you can be an authority on merging it. That said, even then I want a policy that thrid eyes should look at it first.

**Comment by Eirenliel** [comment]
I added a bunch of owners and rules and removed the system with LGTM, instead @TheButlah, @Louka3000, I and @ButterscotchV are now maintainers and can merge if branch protection rules are satisfied. Which I changed to require 2 reviews and code owner reviews.

**Comment by Louka3000** [comment]
From Discord:
- All code owners for the PR must approve (forced by GH)
- Need at least 2 total approving reviews by anyone in the GH contributor team (forced by GH)
- Need checks to pass and no change requests (forced by GH)
- No change request for a day (not forced by GH)

**Review by Louka3000 — APPROVED** [review]

**Review by ImUrX — APPROVED** [review]

**Review by TheButlah — APPROVED** [review]

**Review by ButterscotchV — APPROVED** [review]

**Pull request merged**
